src/daily-quest/suggestion/suggestion.service.ts

```diff
@@ -0,0 +1,172 @@
+import { prisma } from "@/db/prisma";
+import {
+  NotFoundDomainException,
+  UniqueConstraintDomainException,
+  UnknownDomainException,
+} from "@/errors/domain";
+import {
+  DailyQuestSuggestionModel,
+  toDailyQuestSuggestionModel,
+} from "@/daily-quest/suggestion/suggestion.domain.types";
+import QueryParams from "@/types/queryParams";
+import { PrismaClientKnownRequestError } from "@prisma/client/runtime/library";
+import prismaErrorCodes from "@/types/prismaErrorCodes";
+import { logPrismaKnownError } from "@/utils/logger";
+import {
+  CreateDailyQuestSuggestionFormType,
+  UpdateDailyQuestSuggestionFormType,
+} from "@/daily-quest/suggestion/suggestion.forms";
+import {
+  validateDailyQuestSuggestionExists,
+  validateDailyQuestSuggestionForUpdate,
+  validateDailyQuestSuggestionWithTitleIconDoesntExists,
+} from "@/daily-quest/suggestion/suggestion.validators";
+import DailyQuestService from "@/daily-quest/daily-quest.service";
+import InconsistentColumnDataDomainException from "../../errors/domain/inconsistentColumnDataDomain";
+
+export default class DailyQuestSuggestionService {
+  private dailyQuestService = new DailyQuestService();
+
+  /**
+   * List daily quests suggestions excluding active daily quests by given user
+   */
+  findAll = async (
+    userId: string,
+    { limit, sortBy, sortOrder }: QueryParams
+  ): Promise<DailyQuestSuggestionModel[]> => {
+    const idsToExclude =
+      await this.dailyQuestService.getDailyQuestIdsAddedWithSuggestion(userId);
```

Comment:
I don't like the name getDailyQuestIdsAddedWithSuggestion, but I have no ideas how to improve that. Just not very clear what is going on

Reply:
This exact method calls `getDailyQuestsAddedWithSuggestion` and then just maps the result to an array of suggestion IDs.

`getDailyQuestsAddedWithSuggestion` gets all DailyQuests from DB that were added from a DailyQuests suggestion list:
<img width="262" alt="image" src="https://github.com/user-attachments/assets/c881c246-8655-4c04-b68a-3dd2f4aebdcd" />

So maybe `getDailyQuestsAddedFromSuggestion` and `getDailyQuestIdsAddedFromSuggestion` or `getDailyQuestsFromSuggestion` and `getDailyQuestIdsFromSuggestion`?